[Lens] speed up functional test suite by setting the time picker via ui settings

mbondyra opened the pull request:
1. Speeds up functional test suite by setting the time picker via ui settings
2. removes legacy tests from editor frame (replaced by adding some conditions to drag_drop functional tests) as they were not the best tests anyway

mbondyra committed:
[Lens] speed up functional test suite by setting the time picker via ui settings

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/246983)
* Commit: edbf358cdd0af627918b7276bc01920a6e610549

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246983","number":246983,"commit":"edbf358cdd0af627918b7276bc01920a6e610549"}],"number":246983}
buildkite-pr-comment-->

dej611 commented on x-pack/test/functional/apps/lens/group1/smokescreen.ts:
```diff
@@ -20,10 +20,12 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
   const config = getService('config');
 
   describe('lens smokescreen tests', () => {
+    before(async () => {
```
Can you add an `after` task to restore it back when done with the `await timePicker.resetDefaultAbsoluteRangeViaUiSettings();`?

Alternatively this could be applied at `index.ts` level and apply for all suites within that group - I think it is already like that for a couple of groups (see group1 for an example )

mbondyra committed:
remove non-needed code

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/248842)
* Commit: 8665398d2365651acf16a0db2b1cd0da67263d76

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/248842#019301c3-858e-4deb-b2a1-599c04192f88)

### History
* :green_heart: [Build #246983](https://buildkite.com/elastic/kibana-pull-request/builds/246983) succeeded edbf358cdd0af627918b7276bc01920a6e610549

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248842","number":248842,"commit":"8665398d2365651acf16a0db2b1cd0da67263d76"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/246983","number":246983,"commit":"edbf358cdd0af627918b7276bc01920a6e610549"}],"number":248842}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/248892)
* Commit: fdbee853b8634e21bd2659f1498f1574730e1d4a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/248892#0193021c-a37a-4c7d-8895-689745698a6f)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248892","number":248892,"commit":"fdbee853b8634e21bd2659f1498f1574730e1d4a"}],"number":248892}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/248909)
* Commit: cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7

### Failed CI Steps
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/248909#0193024c-5d17-4805-8864-4486c43e558c)
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/248909#0193025d-29df-4db0-a5b2-c7531ae7ed47)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/248909#0193024c-5d17-4805-8864-4486c43e558c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/248909/jobs/0193024c-5d17-4805-8864-4486c43e558c/artifacts/0193025d-16ef-4d5a-adb7-97c764fb6986) FTR Configs #<span></span>14 / lens app - Agg based Vis Open in Lens Pie should convert terms to slice by
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/248909#0193025d-29df-4db0-a5b2-c7531ae7ed47) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/248909/jobs/0193025d-29df-4db0-a5b2-c7531ae7ed47/artifacts/01930266-d4a8-49fb-9bf4-656e638bac6b) FTR Configs #<span></span>14 / lens app - Agg based Vis Open in Lens Pie should convert terms to slice by

### History
* :broken_heart: [Build #248892](https://buildkite.com/elastic/kibana-pull-request/builds/248892) failed fdbee853b8634e21bd2659f1498f1574730e1d4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248909","number":248909,"commit":"cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248892","number":248892,"commit":"fdbee853b8634e21bd2659f1498f1574730e1d4a"}],"number":248909}
buildkite-pr-comment-->

Commit:
[Fleet] Added default index pattern creation to stream-based installation (#199122)

**Related to: https://github.com/elastic/kibana/pull/195888**

## Summary

Add default index pattern creation to the new stream-based package
installation method to match the behavior of standard package
installation.

Switching to stream-based package installation resulted in the default
index patterns not being created, even after installing the rules
package. While this likely doesn’t affect production, as multiple
integrations are usually installed in Kibana (creating the default index
pattern in any case), this change has impacted some tests:
https://github.com/elastic/kibana/pull/199030. So restoring the original
behaviour

flash1293 committed:
[Observability Onboarding] fix docker integration double detection (#199237)

Closes https://github.com/elastic/kibana/issues/195912

See:
https://github.com/elastic/kibana/issues/195912#issuecomment-2460870604

walterra committed:
[ML] Migrate ColorRangeLegend from SCSS to emotion. (#199156)

## Summary

Part of https://github.com/elastic/kibana/issues/140695

Migrates SCSS to emotion for ColorRangeLegend.

<img
src="https://github.com/user-attachments/assets/3444fa49-2ef2-4706-b15e-c3e8415c6293"
/>

<img
src="https://github.com/user-attachments/assets/a79648a5-1230-41b6-b8f5-c1ff2dc9f996"
width="75%" />

### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)

shahzad31 committed:
[Synthetics] Refactor bulk delete monitor and params routes !! (#195420)

## Summary

Refactor bulk delete monitor and params routes !! 

We need to remove usage for body from DELETE route.

### Params

Params can be bulk delete now with POST request to
`/params/_bulk_delete` endpoint

### Monitors
Monitors can be bulk delete now with POST request to
`/monitors/_bulk_delete` endpoint

maryam-saeidi committed:
Fix getAlertSummary returning 400 (Bad Request) (#199116)

Fixes #190126
Fixes #184649

## Summary

This PR fixes alert summary API returning 400 when there is no featured
by only sending this request when featureId is available, otherwise
showing a default state as shown below:

#### Alert details page

|With featureId | Without featureId (intermediate state)|
|---|---|

|![image](https://github.com/user-attachments/assets/d4b4be5a-43be-4365-bd28-66c9dd53d979)|![image](https://github.com/user-attachments/assets/be437052-7c94-4d89-9238-b63264a812c8)|

#### Rule details page
|With featureId | Without featureId|
|---|---|

|![image](https://github.com/user-attachments/assets/28ff4ac5-ffc7-4c9e-8087-408f2ccff98e)|![image](https://github.com/user-attachments/assets/b82f530e-a27c-44e5-a241-c179bc8f1f46)|

tiansivive committed:
[Entity Store] Aligning mappings with ECS (#199001)

## Summary

This PR corrects some of the mappings set in the component template to
follow ECS guidelines.

#### How to test

Initialise an entity engine via devtools:
```
POST kbn:/api/entity_store/engines/host/init
{}
```

Check the mappings with:
```
GET .entities.v1.latest.security_<entityType>_default/_mapping
```

logeekal committed:
[Security Solution] Deletes Old Timeline Code (#196243)

## Summary

Fixes https://github.com/elastic/security-team/issues/9676

This is follow up of the PR :
https://github.com/elastic/kibana/pull/195959 where we had removed
feature flag `unifiedComponentsInTimelineDisabled`.

This PR completes unified components migration in Timeline. 

## Changes

Most changes in this PR are : 

1. Deletion for the obsolete code.
2. Refactoring of needed utils ( only 3. have marked in comments below )
3. Updating import path of above utils.

maryam-saeidi committed:
Use getUrlPartsWithStrippedDefaultPort instead of getUrlParts (#199264)

## Summary

Use getUrlPartsWithStrippedDefaultPort to avoid this
[issue](https://github.com/elastic/kibana/pull/195902#issuecomment-2411667499)
on MKI:

```
Serverless Observability - Deployment-agnostic api integration - Custom Threshold rule AVG - PCT - FIRED Rule creation should set correct action variables

Error: expected 'https://bk-serverless-ftr-3067-e697d43e3ad9-e0ac80.kb.eu-west-1.aws.qa.elastic.cloud/app/observability/alerts/1e0c2d3e-e5c2-4bfe-9df0-d46681253b9f'
to sort of equal 'https://bk-serverless-ftr-3067-e697d43e3ad9-e0ac80.kb.eu-west-1.aws.qa.elastic.cloud:443/app/observability/alerts/1e0c2d3e-e5c2-4bfe-9df0-d46681253b9f'
```

jcger committed:
[Response Ops][Connectors] Refactor Jira Connector to use latest API only (#197787)

## Summary

Jira Cloud and Datacenter work using the same API urls. In this PR we
remove the calls to the capabilities API which was being used to know
the API url we needed to hit

To test it:
- Create Jira Cloud and Datacenter connectors
- Test all use cases related to them

Related to https://github.com/elastic/kibana/issues/189017

## Research Work

**getCapabilities, createIncident and getIncident** are always the same,
therefore ignored for the rest of this document

- getCapabilities: `/rest/capabilities`
- createIncident: `/rest/api/2/issue`
- getIncident: `/rest/api/2/issue`

## API links

- Cloud:
https://developer.atlassian.com/cloud/jira/platform/rest/v2/intro/#version
- DC: https://docs.atlassian.com/software/jira/docs/api/REST/9.17.0/

### Expected API urls based on the API links

- Get issue types

- Cloud: `GET /rest/api/2/issue/createmeta/{projectIdOrKey}/issuetypes`
  - DC:`GET /rest/api/2/issue/createmeta/{projectIdOrKey}/issuetypes`

- Get fields by issue type
- Cloud: `GET
/rest/api/2/issue/createmeta/{projectIdOrKey}/issuetypes/{issueTypeId}`
- DC:
`GET /rest/api/2/issue/createmeta/{projectIdOrKey}/issuetypes/{issueTypeId}`

### API we hit

- Get issue types

- Cloud `GET
/rest/api/2/issue/createmeta?projectKeys=ROC&expand=projects.issuetypes.fields`
(variable name we are using is `getIssueTypesOldAPIURL`)
- DC `GET /rest/api/2/issue/createmeta/RES/issuetypes` (variable name is
`getIssueTypesUrl`)

- Get fields by issue type
- Cloud `GET
/rest/api/2/issue/createmeta?projectKeys=ROC&issuetypeIds={issueTypeId}&expand=projects.issuetypes.fields`
(variable name is `getIssueTypeFieldsOldAPIURL`)
- DC `GET /rest/api/2/issue/createmeta/RES/issuetypes/{issueTypeId}`
(variable name is `getIssueTypeFieldsUrl`)

#### Analysed use cases to retrieve API urls we hit

- created a case with JIRA Cloud as Connector
- did a connector test with JIRA Cloud as connector
- created a case with JIRA DC as connector
- did a connector test with JIRA DC as connector

### Conclusions

- We are not using the right endpoints for Cloud, we should update them
to use the same endpoints.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Christos Nasikas <christos.nasikas@elastic.co>
Co-authored-by: adcoelho <antonio.coelho@elastic.co>
Co-authored-by: Antonio <antoniodcoelho@gmail.com>
Co-authored-by: Lisa Cawley <lcawley@elastic.co>

SoniaSanzV committed:
Allow empty spaces un Gsub processor (#197815)

Closes [#191920](https://github.com/elastic/kibana/issues/191920)

## Summary
In the pattern field of the GSUB processor, we were validating that the
field is not empty. But for this case we must allow spaces as values. To
fix this, I've added a second parameter `trimString` to the `emptyField`
validator. If the parameter is set to true by default so the behavior of
the validator doesn't change in all the fields that already use it. But
when the parameter is set to false, as in this case, the entry value
string is not trimmed, so it only validate to false if the field is
actually empty. I also added some unit tests to make sure that this new
condition doesn't break the validator.


https://github.com/user-attachments/assets/049f7424-00c6-4bb2-ab89-2ce158ac4c4e



#### Notes
In the video it can be seen that a banner flashes when saving. It is a
bug that already happened before this one. I've opened [an
issue](https://github.com/elastic/kibana/issues/197810) to address it.

ppisljar committed:
pass elasticsearch response to browser as a stream (#193060)

Remove decoding and re-encoding of the ES-response from Kibana-server.
Resolves https://github.com/elastic/kibana/issues/189640

jloleysens committed:
[OAS] Delete (empty) staging files again (#199152)

mbondyra committed:
withdraw failing tests

mbondyra committed:
withdraw failing tests

js-jankisalvi committed:
[ResponseOps][Connectors] Allow to use POST method for get case information in case management webhook (#197437)

## Summary

Resolves https://github.com/elastic/kibana/issues/178074

This PR allows to use POST method and JSON payload for body for get case
information in case management webhook.

<img width="1279" alt="Screenshot 2024-10-24 at 15 02 33"
src="https://github.com/user-attachments/assets/aaabc5b8-cd9e-46d6-aab8-85a27731bd88">

### How to test
- Create a case management webhook connector [as per
documentation](https://www.elastic.co/guide/en/kibana/master/cases-webhook-action-type.html)
- Use any public API which supports GET and POST methods 
- Use POST method for Get case information
- Verify that it validates the URL and JSON payload correctly 
- Test the connector using Test tab
- Create a rule with action using case management webhook connector
- Verify alerts are generated and action executed without any errors

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))

ymao1 committed:
Fixes flaky backfill tests. (#198592)

Resolves https://github.com/elastic/kibana/issues/192144
Resolves https://github.com/elastic/kibana/issues/198168
Resolves https://github.com/elastic/kibana/issues/197239

## Summary

Minor fixes to the backfill functional tests to try to reduce flakiness.

viduni94 committed:
[Obs AI Assistant] Update the word "chat" to "conversation" across the UI

Closes https://github.com/elastic/kibana/issues/196999

## Summary

### Problem
The word "chat" and "conversation" have been used in different parts of
the UI to refer to the same thing, causing some inconsistency.

### Solution
Update all occurrences of "chat" to "conversation"

mbondyra commented:
/ci

dkirchan committed:
[Security Solution][Serverless] Github tickets / notifications (#197265)

## Summary

This PR is accompanied with the [PR in kibana-operations
repo](https://github.com/elastic/kibana-operations/pull/236) and
integrates the security solution quality gate pipelines with the
functionality to create ticket when a test fails and then notify the
respective **codeowner** team with a _Test failed alert_. Also when a
test fails a second time and a ticket exist a new comment is added in
the same issue that the test failed again in a given pipeline.

Last a similar flow exist when a test is skipped by a team. 

**Specifications:**

- In the `failed_test_reporter_cli.ts` [a new field is introduced
](https://github.com/elastic/kibana/compare/main...security-mki-github-tickets#diff-bb801c18fd2e1a3a36a3b39fbf02c1abe337c46c201ad5a01239a9d2501d4b56R47)`prependTitle`
which is initialized by the environmental variable
`process.env.PREPEND_FAILURE_TITLE` if exists, in order to add a custom
text of preference before the issue title.

The scope of this is to be able to give the team the opportunity to add
some tags or any convention agreed within the team in the issue created
in order to easily understand the context without opening the ticket. If
this field is not initialized, the normal existing flow proceeds.
[Here](https://github.com/elastic/kibana/issues/197133) an example of
the prependTitle usage can be seen where the tags `[MKI][QA]` are
prepended in order to identify where did the test fail, having the same
exactly tests running on both CI and MKI. This means that a github issue
with the exact same title would be created for both cases if this
prepend title field would not exist.

- In the
[junit_transformer/lib.ts](https://github.com/elastic/kibana/compare/main...security-mki-github-tickets#diff-31c5651af613c7d02139f3e9fccd00ddb997f2502523372dd19db9e0659a66d6R278)
a new functionality is introduced to cover an existing issue. The
existing issue is: the fact that we are retrying the failed test in
cypress in the `parallel_serverless` script, leads to two junit xml
result files completely identical with only difference the execution
time and the timestamps. This change will take one by one the xml
outputs, transform them exactly as it was happening before and then save
the file, but also remove the time and timestamp fields, convert it to a
string and add it to a "state" list. Then in a next file if it is
already parsed and saved to the list having the exact same results
within, it deletes the file instead of saving it transformed.

The problem before this fix was the fact that when two xml outputs
existed, a github ticket was created and when parsing-uploading the
second file it was immediately failing for a second time as well. This
means that `new-failure` notification was never triggered after the
github actions flow split, and the existing failure was always
triggered, something that most teams have disabled.

With the fix, the identical files are parsed but only once uploaded and
the new failure flow works again.

vadimkibana committed:
[ES|QL] Support params in column nodes and function names (#198957)

## Summary

Closes https://github.com/elastic/kibana/issues/198251

Adds support for the new ES|QL param syntax in the following places:

- Support all param types in `column` AST node types (single level and
nested)
- Support those new language features in:
   - AST
   - Parser
   - Traversal: `Walker`
- Support in validation and autocomplete:
  - Makes tests pass
  - Make validation and autocomplete not complain on param usage in
    - Function names
    - Function arguments
    - In function names where aggregation functions expected
    - Column names
      - First level
      - Nested
      - Exception out of "unknownField" errors


### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### For maintainers

- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)

---------

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/249202)
* Commit: 0c51d510475a9b3fe5e67aba97198747ca78c757

### Failed CI Steps
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306da-49f8-40fe-a284-622e839a2ee4)
* [FTR Configs #15](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306da-49f9-4965-adb4-891dffed9bf2)
* [FTR Configs #15](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306f7-e0a0-4c80-88e4-af60df3b2c67)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306da-49f8-40fe-a284-622e839a2ee4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249202/jobs/019306da-49f8-40fe-a284-622e839a2ee4/artifacts/019306f5-ada7-4c21-aa97-994082cbdda7) FTR Configs #<span></span>14 / lens app - Agg based Vis Open in Lens XY should convert split series
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306da-49f9-4965-adb4-891dffed9bf2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249202/jobs/019306da-49f9-4965-adb4-891dffed9bf2/artifacts/019306f7-d11e-4368-9f8b-f121b38285bd) FTR Configs #<span></span>15 / lens app - group 2 lens datatable should able to sort a table by a column
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249202#019306f7-e0a0-4c80-88e4-af60df3b2c67) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249202/jobs/019306f7-e0a0-4c80-88e4-af60df3b2c67/artifacts/01930705-fcd4-4073-9ebd-0c5471f12879) FTR Configs #<span></span>15 / lens app - group 2 lens datatable should able to sort a table by a column

### History
* :broken_heart: [Build #249120](https://buildkite.com/elastic/kibana-pull-request/builds/249120) failed 0e808a2b33423dc0a966a1d435b5d4e040fbe338
* :broken_heart: [Build #248909](https://buildkite.com/elastic/kibana-pull-request/builds/248909) failed cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7
* :broken_heart: [Build #248892](https://buildkite.com/elastic/kibana-pull-request/builds/248892) failed fdbee853b8634e21bd2659f1498f1574730e1d4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249202","number":249202,"commit":"0c51d510475a9b3fe5e67aba97198747ca78c757"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249120","number":249120,"commit":"0e808a2b33423dc0a966a1d435b5d4e040fbe338"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248909","number":248909,"commit":"cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248892","number":248892,"commit":"fdbee853b8634e21bd2659f1498f1574730e1d4a"}],"number":249202}
buildkite-pr-comment-->

opauloh committed:
[Agentless] Fix agentless global_data_tags (#199247)

shahzad31 committed:
[Synthetics] Fix location remove via public API !! (#199170)

## Summary

Fixes https://github.com/elastic/kibana/issues/199108

Fix location remove via public API !!

### Testing 

Create a Monitor that is configured in two locations:
```
POST kbn:/api/synthetics/monitors
{
  "type": "http",
  "name": "healthcheck",
  "url": "https://www.elastic.co",
  "locations": ["united_kingdom", "germany"]
}
```
Extract from the response the config_id
Update the monitor to remove one location
```
PUT kbn:/api/synthetics/monitors/<config-id>
{
  "type": "http",
  "name": "healthcheck",
  "url": "https://www.elastic.co",
  "locations": ["united_kingdom"]
}
```


This should remove Germany location from monitor !!

machadoum committed:
[SecuritySolution] [EntityStore] Use SecuritySolution link to prevent page refresh (#199278)

## Summary

Use in-app navigation to prevent page refresh.



https://github.com/user-attachments/assets/cdb1ab08-62ab-4e51-b92f-2afafa56476d

stratoula committed:
[ES|QL] Improves the keep command description (#199312)

## Summary

Closes https://github.com/elastic/kibana/issues/199297

<img width="1167" alt="image"
src="https://github.com/user-attachments/assets/c26d07e8-f782-412c-968c-c551a9bc7942">

Bamieh committed:
[i18n] Integrate 8.16.0 Translations (#199145)

Integrating latest translations extracted from main branch.
Skipping backports from main to target branches since the `i18n_check`
might trim unused translations that are still used in different
branches. Integration script is ran against each target branch
separately.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/249259)
* Commit: 5173f8daca07ac01238c58dbb0a50e7b2d8fa9fb

### Failed CI Steps
* [FTR Configs #5](https://buildkite.com/elastic/kibana-pull-request/builds/249259#01930731-69a9-4f99-9c22-2e7c6d540697)
* [FTR Configs #5](https://buildkite.com/elastic/kibana-pull-request/builds/249259#0193074e-e6c9-451a-9674-ec4ccec918f9)
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/249259#01930731-69ab-43b2-8be8-0c7856988897)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249259#01930731-69ab-43b2-8be8-0c7856988897) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249259/jobs/01930731-69ab-43b2-8be8-0c7856988897/artifacts/0193074f-4af4-45d5-84ff-2957e4d97bab) FTR Configs #<span></span>6 / lens app - Agg based Vis Open in Lens XY should convert split series
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249259#01930731-69a9-4f99-9c22-2e7c6d540697) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249259/jobs/01930731-69a9-4f99-9c22-2e7c6d540697/artifacts/0193074e-c35a-4a66-bec3-fafe172db8eb) FTR Configs #<span></span>5 / lens app - group 2 lens datatable should able to sort a table by a column
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/249259#0193074e-e6c9-451a-9674-ec4ccec918f9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/249259/jobs/0193074e-e6c9-451a-9674-ec4ccec918f9/artifacts/0193075d-641f-4b62-836b-384dfde0b827) FTR Configs #<span></span>5 / lens app - group 2 lens datatable should able to sort a table by a column

### History
* :broken_heart: [Build #249202](https://buildkite.com/elastic/kibana-pull-request/builds/249202) failed 0c51d510475a9b3fe5e67aba97198747ca78c757
* :broken_heart: [Build #249120](https://buildkite.com/elastic/kibana-pull-request/builds/249120) failed 0e808a2b33423dc0a966a1d435b5d4e040fbe338
* :broken_heart: [Build #248909](https://buildkite.com/elastic/kibana-pull-request/builds/248909) failed cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7
* :broken_heart: [Build #248892](https://buildkite.com/elastic/kibana-pull-request/builds/248892) failed fdbee853b8634e21bd2659f1498f1574730e1d4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249259","number":249259,"commit":"5173f8daca07ac01238c58dbb0a50e7b2d8fa9fb"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249202","number":249202,"commit":"0c51d510475a9b3fe5e67aba97198747ca78c757"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249120","number":249120,"commit":"0e808a2b33423dc0a966a1d435b5d4e040fbe338"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248909","number":248909,"commit":"cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248892","number":248892,"commit":"fdbee853b8634e21bd2659f1498f1574730e1d4a"}],"number":249259}
buildkite-pr-comment-->

elastic-renovate-prod[bot] committed:
Update dependency @redocly/cli to ^1.25.10 (main) (#199300)

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>

nchaulet committed:
[Fleet] Ignore missing policies when fetching agent data (#199325)

michaelolo24 committed:
[WIP] - Unskip threat hunting tests (#195662)

1. Added `cypress-network-idle` to better wait for a full page load as a
loading indicator flicker could cause flakiness.

2. List of addressed open issues:

Resolves https://github.com/elastic/kibana/issues/191819
Resolves https://github.com/elastic/kibana/issues/178367
Resolves https://github.com/elastic/kibana/issues/176757
Resolves https://github.com/elastic/kibana/issues/176759
Resolves https://github.com/elastic/kibana/issues/176758
Resolves https://github.com/elastic/kibana/issues/193840
Resolves https://github.com/elastic/kibana/issues/182206
Resolves https://github.com/elastic/kibana/issues/181977
Resolves https://github.com/elastic/kibana/issues/168768
Resolves https://github.com/elastic/kibana/issues/168769
Resolves https://github.com/elastic/kibana/issues/168770
Resolves https://github.com/elastic/kibana/issues/168771
Resolves https://github.com/elastic/kibana/issues/168772
Resolves https://github.com/elastic/kibana/issues/183104
Resolves https://github.com/elastic/kibana/issues/173854

Successful Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7267

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

lukasolson committed:
Move consumers off bsearch endpoint (#196962)

## Summary

Part of https://github.com/elastic/kibana/issues/186139.

Moves direct consumers off of the `internal/bsearch` endpoint and onto
the `internal/search`.

This is in preparation for removing the `internal/bsearch` endpoint
entirely.

Updates automated tests to mock/use the correct service moving forward.

### Checklist

- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Dzmitry Lemechko <dzmitry.lemechko@elastic.co>

alvarezmelissa87 committed:
[ML][AIOPS] Log rate analysis: add basic functional test  (#199054)

This PR adds a basic functional test for
[this](https://github.com/elastic/kibana/pull/198179) change that adds a
switch to Rerun analysis copy if the search bar search changed.

Flaky test runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7344

## Summary

Summarize your PR. If it involves visual changes include a screenshot or
gif.


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

PhilippeOberti committed:
[Security Solution][Alert details] - open related case in new browser tab (#197494)

## Summary

This PR makes a very small change to the alerts details expandable
flyout Insights Correlations tab where we display all related cases in a
table. The first column of that table renders a link to the cases
details page.
When clicking on that link, instead of navigating to the cases details
page with the same browser tab, we know open a new tab, so we don't
loose the context of the alert. The UI displays the icon that we use in
other places to inform users that the action is opening a new tab.

Old behavior

https://github.com/user-attachments/assets/48c68608-9091-45f4-a16b-a19f0af12f87

New behavior

https://github.com/user-attachments/assets/78cf6b81-5044-4f26-9808-8fa8d829d723

https://github.com/elastic/security-team/issues/7034

## Edit

This PR now also makes a change to the alert cases guided onboarding
tour, as opening a new tab breaks the tour. The changes are pretty small
and consist of removing the last step that was forcing the users to
click on the alert details flyout expanded Insights Correlations related
cases section. The tour now ends after creating a case, with a message
saying that cases can be accessed via the toast or via the related cases
in the flyout.

https://github.com/user-attachments/assets/0061fdce-5d43-45a3-8166-f812eaffa7b3

kapral18 committed:
[Security Solution][Data Quality Dashboard][Serverless] add start/end time support for latest_results (#199248)

addresses #191053

- Introduce `defaultStartTime` and `defaultEndTime` props across data
quality context and panels for fetching latest_results and align them
with serverless default time range of last week
- Update hooks to handle new time parameters and include them in storage
results queries.
- Modify server-side helpers and routes to process and filter indices
based on the provided time range.
- Update related tests to accommodate the new time parameters.

Commit:
[Reporting] update puppeteer to version 23.7.0 (#199304)

sphilipse committed:
[AI Assistant] Enable Search in Solution View (#199137)

## Summary

This enables the Search solution in the Observability solution view, so
that connectors (which are required for the AI Assistant) are available.

jbudz committed:
[ci] Add retry for vm image cache updates (#199320)

To address occasional transient network errors.

lcawl committed:
[OpenAPI][DOCS] Capitalize roles tag in serverless overlay (#198883)

kyracho committed:
Increase GCP button size to match Azure and AWS (#199228)

## Summary

Hi! this PR deals with issue #198878 .

The `RadioGroup` size for GCP was changed from `"s"` to `"m"` to match
the buttons for Azure and AWS.

Thanks!

<img width="751" alt="Screenshot 2024-11-06 at 12 39 09 PM"
src="https://github.com/user-attachments/assets/43d9c51b-1519-4065-90ca-a957a8cdbae1">
<img width="746" alt="Screenshot 2024-11-06 at 12 39 20 PM"
src="https://github.com/user-attachments/assets/dced0d76-9430-43ea-a0dd-141665def411">



### Checklist
- [x] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))

(https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### For maintainers
- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)
- [ ] This will appear in the **Release Notes** and follow the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine committed:
skip failing test suite (#192128)

mistic committed:
skip flaky suite (#199363)

mistic committed:
skip flaky suite (#172326)

mistic committed:
skip flaky suite (#184600)

mistic committed:
skip flaky suite (#198109)

ashokaditya committed:
[DataUsage][Serverless] Add missing tests (#198007)

## Summary

Adds missing UX/API tests for changes added in 
- https://github.com/elastic/kibana/pull/193966
- https://github.com/elastic/kibana/pull/197056
- https://github.com/elastic/kibana/pull/195556
- https://github.com/elastic/kibana/pull/196559


### Checklist
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

TinaHeiligers committed:
Changes deprecated SO HTTP APIs deprecation field to object (#197936)

fix https://github.com/elastic/kibana/issues/197721.

The route deprecation field changed from a boolean to an object, where
the object contains information that is used in deprecation issues that
the Upgrade Assistant shows.

This PR makes the necessary changes in the deprecated Saved Objects HTTP
APIs.

This PR also includes a release notes entry for the API deprecations
that was missing.
![Screenshot 2024-10-29 at 12 01
29](https://github.com/user-attachments/assets/5c47c697-fbae-4b2e-8c6c-cd4701a667df)

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### How to test this:

- Pull PR, run es against a trial license and start kibana as usual.
- Make a curl request to Kibana to get the config saved object: 

```
curl --location 'localhost:5601/abc/api/saved_objects/config/9.0.0' \
--header 'Content-Type: application/json' \
--header 'Accept-Encoding: gzip, deflate, br' \
--header 'kbn-xsrf: kibana' \
--header 'Kbn-Version: 9.0.0' \
--header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ=='
```

- Navigate to Upgrade Assistant and observe Kibana has at least 1
deprecation warning.
- View Kibana's warnings, you should see a warning entry for `The "GET
/api/saved_objects/{type}/{id}" route is deprecated`

![Screenshot 2024-11-06 at 16 26
26](https://github.com/user-attachments/assets/3b6a5644-3e5e-403e-a0f6-015686675b9f)

- click on the deprecation and you should see more detail about the
deprecated API that's been used in the flyout:

![Screenshot 2024-11-06 at 16 26
44](https://github.com/user-attachments/assets/696aaf8f-fb6b-4c61-bc3c-b3745f85059a)

- resolve the deprecation warning
- Kibana should continue to issue requests to the deprecated SO HTTP
APIs because these APIs have not been removed yet.


### Risk Matrix

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| End user concern from the deprecation warning that says the routes
have been deprecated | Low | Low | The APIs have been deprecated since
8.7 and recommends using public APIs instead. |

### For maintainers

- [x] This will appear in the **Release Notes** and follow the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
Unauthorized route migration for routes owned by kibana-security (#198334)

### Authz API migration for unauthorized routes

This PR migrates unauthorized routes owned by your team to a new
security configuration.
Please refer to the documentation for more information: [Authorization
API](https://docs.elastic.dev/kibana-dev-docs/key-concepts/security-api-authorization)

### **Before migration:**
```ts
router.get({
  path: '/api/path',
  ...
}, handler);
```

### **After migration:**
```ts
router.get({
  path: '/api/path',
  security: {
    authz: {
      enabled: false,
      reason: 'This route is opted out from authorization because ...',
    },
  },
  ...
}, handler);
```

### What to do next?
1. Review the changes in this PR.
2. Elaborate on the reasoning to opt-out of authorization.
3. Routes without a compelling reason to opt-out of authorization should
plan to introduce them as soon as possible.
2. You might need to update your tests to reflect the new security
configuration:
  - If you have snapshot tests that include the route definition.

## Any questions?
If you have any questions or need help with API authorization, please
reach out to the `@elastic/kibana-security` team.

---------

Co-authored-by: “jeramysoucy” <jeramy.soucy@elastic.co>
Co-authored-by: Larry Gregory <larry.gregory@elastic.co>
Co-authored-by: Elena Shostak <165678770+elena-shostak@users.noreply.github.com>
Co-authored-by: Elena Shostak <elena.shostak@elastic.co>

cqliu1 committed:
[Canvas] Remove expression lifecycle docs (#199006)

## Summary

Closes https://github.com/elastic/kibana/issues/193475.

This removes outdated Canvas expression docs that describes capabilities
of the expression that are no longer true.


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature
release.

When forming the risk matrix, consider some of the following examples
and how they may potentially impact the change:

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space.
| Low | High | Integration tests will verify that all features are still
supported in non-default Kibana Space and when user switches between
spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions
when multiple Kibana nodes are polling for the same tasks. | High | Low
| Tasks are idempotent, so executing them multiple times will not result
in logical error, but will degrade performance. To test for this case we
add plenty of unit tests around this logic and document manual testing
procedure. |
| Code should gracefully handle cases when feature X or plugin Y are
disabled. | Medium | High | Unit tests will verify that any feature flag
or plugin combination still results in our service operational. |
| [See more potential risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)
- [ ] This will appear in the **Release Notes** and follow the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

shahzad31 committed:
[Synthetics] Handle private locations simultaneous edits !! (#195874)

## Summary

Fixes https://github.com/elastic/kibana/issues/190801 !!

Handle private locations simultaneous edits !! 

Registered a new saved object to handle private locations properly.
Instead of using a singleton, now each private location will be
represented by it's own saved object.

### Existing private locations
When we are doing any write operation, we migrate them to new kind of
saved object and remove the legacy saved object type.


### Testing

- Create multiple private locations on main 
- Switch to branch and create few more locations
- Make sure all private locations are editable, deleteable and have been
migrated to new saved object types in the course

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[APM] Skip transaction details test and removing tutorial tests (#199336)

It is what it is!

marius-dr committed:
[QA]Change screenshot size for metricbeat dashboard test (#199378)

## Summary

Change the default screenshot compare size for the metricbeat dashboard.

kibanamachine committed:
Authorized route migration for routes owned by @elastic/kibana-core (#198187)

Co-authored-by: Alejandro Fernández Haro <alejandro.haro@elastic.co>

kibanamachine committed:
skip failing test suite (#199372)

kibanamachine committed:
skip failing test suite (#196226)

kibanamachine committed:
skip failing test suite (#199356)

yansavitski committed:
[Search] [Playground] [Bug] Add Search Playgrounnd to Serverless manifest (#199350)

Missed searchPlayground in kibana manifest

kibanamachine committed:
[api-docs] 2024-11-08 Daily api_docs build (#199415)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/885

kibanamachine committed:
Authorized route migration for routes owned by @elastic/security-detection-engine (#198195)

### Authz API migration for authorized routes

This PR migrates `access:<privilege>` tags used in route definitions to
new security configuration.
Please refer to the documentation for more information: [Authorization
API](https://docs.elastic.dev/kibana-dev-docs/key-concepts/security-api-authorization)

### **Before migration:**
Access control tags were defined in the `options` object of the route:

```ts
router.get({
  path: '/api/path',
  options: {
    tags: ['access:<privilege_1>', 'access:<privilege_2>'],
  },
  ...
}, handler);
```

### **After migration:**
Tags have been replaced with the more robust
`security.authz.requiredPrivileges` field under `security`:

```ts
router.get({
  path: '/api/path',
  security: {
    authz: {
      requiredPrivileges: ['<privilege_1>', '<privilege_2>'],
    },
  },
  ...
}, handler);
```

### What to do next?
1. Review the changes in this PR.
2. You might need to update your tests to reflect the new security
configuration:
  - If you have tests that rely on checking `access` tags.
  - If you have snapshot tests that include the route definition.
- If you have FTR tests that rely on checking unauthorized error
message. The error message changed to also include missing privileges.

## Any questions?
If you have any questions or need help with API authorization, please
reach out to the `@elastic/kibana-security` team.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Nikita Khristinin <nkhristinin@gmail.com>

mbondyra committed:
[Visualize] remove never executing functional tests code (#198512)

## Summary

Removes the code that goes through the condition if the vis comes from
new or legacy library. Pie charts or xy charts don't use vislib
implementation anymore, so the condition will always give the same
result.

georgianaonoleata1904 committed:
[ResponseOps][Cases]Allow users to push to external service when the case is closed (#198146)

Closes https://github.com/elastic/kibana/issues/192552

## Summary

- Removed the restriction that didn't allow user to push a closed case
to an external service.


https://github.com/user-attachments/assets/82c49524-5685-40c0-b025-d6b38ab4e056

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Antonio <antonio.coelho@elastic.co>

florent-leborgne committed:
[Docs] Update Kibana ESQL docs (#199371)

This PR updates ESQL docs currently in the Kibana guide. I'll do another
PR to the Elasticsearch docs to update the docs there as well.

Rel: https://github.com/elastic/platform-docs-team/issues/552

semd committed:
[SecuritySolution][Navigation] Unify Notes link (#199374)

## Summary

Bug Issue: https://github.com/elastic/kibana/issues/197694

Unify the Notes link configuration.

- ESS classic nav (default) -> subLink of the `Manage` section
- ESS solution nav / Serverless -> `Manage` section is removed and the
`Notes` link is relocated as a sublink of the `Investigations` section
  
Breadcrumbs and global search results are now consistent with the page's
location in each environment.

### Screenshots

Classic nav:

![search
classic](https://github.com/user-attachments/assets/0c0f31e8-3e34-4348-b0f0-2f7ea9fc18ab)
![breadcrumbs
classic](https://github.com/user-attachments/assets/1cdeec4b-436f-4fa0-b04d-058fb8802509)

Solution nav:

![search
solution](https://github.com/user-attachments/assets/a915dc1e-ecdc-4d41-97a7-fa280307fb9e)
![breadcrumbs
solution](https://github.com/user-attachments/assets/254b66b2-da1f-4563-acaf-568018035daf)

mbondyra committed:
[Lens] test commit

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/249518)
* Commit: 13478ff6a81fb2eee3260a9ffaeae5e829e10d70

### Failed CI Steps
* [Build and Validate Next Docs](https://buildkite.com/elastic/kibana-pull-request/builds/249518#01930b7e-2815-415a-9850-d2d453ff7a84)

### History
* :broken_heart: [Build #249259](https://buildkite.com/elastic/kibana-pull-request/builds/249259) failed 5173f8daca07ac01238c58dbb0a50e7b2d8fa9fb
* :broken_heart: [Build #249202](https://buildkite.com/elastic/kibana-pull-request/builds/249202) failed 0c51d510475a9b3fe5e67aba97198747ca78c757
* :broken_heart: [Build #249120](https://buildkite.com/elastic/kibana-pull-request/builds/249120) failed 0e808a2b33423dc0a966a1d435b5d4e040fbe338
* :broken_heart: [Build #248909](https://buildkite.com/elastic/kibana-pull-request/builds/248909) failed cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7
* :broken_heart: [Build #248892](https://buildkite.com/elastic/kibana-pull-request/builds/248892) failed fdbee853b8634e21bd2659f1498f1574730e1d4a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249518","number":249518,"commit":"13478ff6a81fb2eee3260a9ffaeae5e829e10d70"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249259","number":249259,"commit":"5173f8daca07ac01238c58dbb0a50e7b2d8fa9fb"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249202","number":249202,"commit":"0c51d510475a9b3fe5e67aba97198747ca78c757"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/249120","number":249120,"commit":"0e808a2b33423dc0a966a1d435b5d4e040fbe338"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248909","number":248909,"commit":"cd6e8ad39fa2a283b45ee2d7e75f3ed2404039e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/248892","number":248892,"commit":"fdbee853b8634e21bd2659f1498f1574730e1d4a"}],"number":249518}
buildkite-pr-comment-->
